### PR TITLE
Update DPS client transport init

### PR DESCRIPTION
**Pull request by drwill-ms**

### CODE REVIEW

File: iothub/device/src/Transport/IotHubClientTransportProtocol.cs [hunk 1]
```diff
@@ -1,9 +1,9 @@
 ﻿namespace Microsoft.Azure.Devices.Client
 {
     /// <summary>
-    /// The protocol over which a transport (i.e., MQTT, AMQP) communicates
+    /// The protocol over which a transport (i.e., MQTT, AMQP) communicates.
     /// </summary>
-    public enum TransportProtocol
+    public enum IotHubClientTransportProtocol
```
Comment (drwill-ms) [inline iothub/device/src/Transport/IotHubClientTransportProtocol.cs:6]: Renaming to make distinct from other libraries.

File: provisioning/device/src/Transports/Amqp/ProvisioningTransportHandlerAmqp.cs [hunk 2]
```diff
@@ -28,39 +28,20 @@ public class ProvisioningTransportHandlerAmqp : ProvisioningTransportHandler
         /// <summary>
         /// Creates an instance of the ProvisioningTransportHandlerAmqp class using the specified fallback type.
         /// </summary>
-        /// <param name="transportFallbackType">The fallback type allowing direct or WebSocket connections.</param>
```
Comment (drwill-ms) [inline provisioning/device/src/Transports/Amqp/ProvisioningTransportHandlerAmqp.cs:31]: I can't tell how the client supported "fallback" behavior, but I want to do away with this concept anyway (in accordance with the hub device client), so this is just specifying Tcp or web socket now.

File: provisioning/device/src/Transports/Amqp/ProvisioningTransportHandlerAmqp.cs [hunk 3]
```diff
@@ -28,39 +28,20 @@ public class ProvisioningTransportHandlerAmqp : ProvisioningTransportHandler
...
-
-        /// <summary>
-        /// Registers a device described by the message.
-        /// </summary>
-        /// <param name="message">The provisioning message.</param>
-        /// <param name="timeout">The maximum amount of time to allow this operation to run for before timing out.</param>
-        /// <returns>The registration result.</returns>
-        public override async Task<DeviceRegistrationResult> RegisterAsync(
-            ProvisioningTransportRegisterRequest message,
-            TimeSpan timeout)
```
Comment (drwill-ms) [inline provisioning/device/src/Transports/Amqp/ProvisioningTransportHandlerAmqp.cs:54]: Deprecating the timeout overloads.

File: provisioning/device/src/Transports/ProvisioningClientTransportProtocol.cs [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    /// <summary>
+    /// The protocol over which a transport (i.e., MQTT, AMQP) communicates.
+    /// </summary>
+    public enum ProvisioningClientTransportProtocol
+    {
+        /// <summary>
+        /// Communicate over TCP using the default port of the transport.
+        /// </summary>
+        /// <remarks>
+        /// For MQTT, this this is 1883.
```
Comment (timtay-microsoft) [inline provisioning/device/src/Transports/ProvisioningClientTransportProtocol.cs:12]:
```suggestion
        /// For MQTT, this port is 8883.
```

8883 I believe

File: provisioning/device/src/Transports/ProvisioningClientTransportProtocol.cs [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+        /// For AMQP, this is 5671.
```
Comment (timtay-microsoft) [inline provisioning/device/src/Transports/ProvisioningClientTransportProtocol.cs:13]:
```suggestion
        /// For AMQP, this port is 5671.
```

**Review by timtay-microsoft — APPROVED** [review]

**Pull request merged**